pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go

```diff
@@ -554,14 +554,29 @@ func handleRangefeedError(ctx context.Context, err error) (rangefeedErrorInfo, e
 	}
 }
 
+// catchup alloc is a catchup scan allocation.
+type catchupAlloc func()
+
+// Release implements limit.Reservation
+func (a catchupAlloc) Release() {
+	a()
+}
+
 func acquireCatchupScanQuota(
 	ctx context.Context, ds *DistSender, catchupSem *limit.ConcurrentRequestLimiter,
 ) (limit.Reservation, error) {
 	// Indicate catchup scan is starting;  Before potentially blocking on a semaphore, take
 	// opportunity to update semaphore limit.
-	ds.metrics.RangefeedCatchupRanges.Inc(1)
 	catchupSem.SetLimit(maxConcurrentCatchupScans(&ds.st.SV))
-	return catchupSem.Begin(ctx)
+	res, err := catchupSem.Begin(ctx)
+	if err != nil {
+		return nil, err
+	}
+	ds.metrics.RangefeedCatchupRanges.Inc(1)
+	return catchupAlloc(func() {
+		ds.metrics.RangefeedCatchupRanges.Dec(1)
```

Comment:
There is another `Dec` for this metric in `singleRangeFeed` func, which is now duplicated. Remove that other `Dec`?